### PR TITLE
Fix/udi 97/export items user does not gave access

### PR DESCRIPTION
**Pull request by SergiiTao**
depends on oat-sa/tao-core#2428

Ticket: https://oat-sa.atlassian.net/browse/UDI-97

There was a possibility to fetch resources without having permissions to them.
I've created a service to fetch resources safely.

### CODE REVIEW

File: model/Export/ExportForm.php [hunk 1]
```diff
@@ -56,108 +57,146 @@ abstract class ExportForm extends tao_helpers_form_FormContainer
...
+        return strtolower(tao_helpers_Display::textCleaner($name, '*'));
+    }
+
+    protected function getOptions(core_kernel_classes_Resource ...$resource): array
+    {
+        $itemService = taoItems_models_classes_ItemsService::singleton();
+        $options = [];
+        $disabledOptions = [];
+        foreach ($resource as $instance) {
+            if ($itemService->hasItemModel($instance, [ItemModel::MODEL_URI])) {
```
Comment (wazelin) [inline model/Export/ExportForm.php]: Early return is preferable here IMO, too much of nesting is happening here
Consider doing `if (!$itemService->hasItemModel($instance, [ItemModel::MODEL_URI])) break;`

Comment (SergiiTao) [reply model/Export/ExportForm.php]: `continue?` or i don't see something?

Comment (wazelin) [reply model/Export/ExportForm.php]: Right, `continue`

File: model/Export/ApipPackageExportHandler.php [hunk 2]
```diff
@@ -57,25 +62,28 @@ public function getLabel()
...
+     *
+     * @return tao_helpers_form_Form
+     * @throws common_Exception
+     * @throws common_exception_Error
      */
     public function getExportForm(core_kernel_classes_Resource $resource)
     {
         if ($resource instanceof core_kernel_classes_Class) {
-            $formData = ['class' => $resource];
+            $formData['items'] = $this->getResourceService()->getAllChildren($resource);
```
Comment (wazelin) [inline model/Export/ApipPackageExportHandler.php]: Not only the form should exclude the items, but the export request handler itself should filter out inaccessible resources.
Right now, I can easily download items, I don't see on that form by calling `/taoItems/ItemExport/index`, extending the list of `instances` manually.

Comment (SergiiTao) [reply model/Export/ApipPackageExportHandler.php]: fixed

Comment (wazelin) [reply model/Export/ApipPackageExportHandler.php]: Right, my bad, misplaced the original comment.
Good job!

**Pull request merged**